superset/charts/api.py

```diff
@@ -446,13 +451,13 @@ def data(self) -> Response:
         else:
             return self.response_400(message="Request is not JSON")
         try:
-            query_context, errors = ChartDataQueryContextSchema().load(json_body)
-            if errors:
-                return self.response_400(
-                    message=_("Request is incorrect: %(error)s", error=errors)
-                )
+            query_context = ChartDataQueryContextSchema().load(json_body)
         except KeyError:
             return self.response_400(message="Request is incorrect")
+        except ValidationError as err:
+            return self.response_400(
+                _("Request is incorrect: %(error)s", error=err.messages)
+            )
```

Comment:
Is `except KeyError` still needed here? Based on the docs, `load` only raises `ValidationError` in 3.0+: https://marshmallow.readthedocs.io/en/stable/_modules/marshmallow/schema.html#Schema.load

Reply:
Yes it seems weird, this may possible come from our custom `post_load` https://github.com/apache/incubator-superset/blob/master/superset/charts/schemas.py#L736 and then from https://github.com/apache/incubator-superset/blob/master/superset/common/query_context.py#L57